akka-cluster/src/main/resources/reference.conf

```diff
@@ -280,6 +280,18 @@ akka {
       verbose-gossip-logging = off
     }
 
+    # Enforce configuration compatibility checks when joining a cluster.
+    # Set to off to allow joining nodes to join a cluster even when configuration incompatibilities are detected.
+    # This is particularly useful for rolling updates on clusters that do not support that feature. Since the old
+    # cluster won't be able to send the compatibility confirmation to the joining node, the joining node won't be able
+    # to 'know' if its allowed to join.
+    enforce-config-compat-on-join = on
```

Comment:
What about this one? Should we configure it by default to `on`?

That means that rolling updates will be blocked on unless the users explicitly set it to `off`.

Reply:
it should be `on` by default